server/sonar-qa-util/src/main/java/org/sonarqube/qa/util/pageobjects/RuleDetails.java

```diff
@@ -19,15 +19,91 @@
  */
 package org.sonarqube.qa.util.pageobjects;
 
-import com.codeborne.selenide.Selenide;
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.SelenideElement;
+import java.util.Locale;
 
 import static com.codeborne.selenide.Condition.text;
+import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Selenide.$;
+import static com.codeborne.selenide.Selenide.$$;
 
 public class RuleDetails {
 
+  public RuleDetails shouldHaveType(String type) {
+    $$(".coding-rules-detail-property").findBy(text(type)).shouldBe(visible);
+    return this;
+  }
+
+  public RuleDetails shouldHaveSeverity(String severity) {
+    $(".coding-rules-detail-property .icon-severity-" + severity.toLowerCase(Locale.ENGLISH)).shouldBe(visible);
+    return this;
+  }
+
+  public RuleDetails shouldHaveNoTags() {
+    $(".coding-rules-detail-tag-list").shouldHave(text("No tags"));
+    return this;
+  }
+
+  public RuleDetails shouldHaveDescription(String description) {
+    $(".js-rule-description").shouldHave(text(description));
+    return this;
+  }
+
   public RuleDetails shouldBeActivatedOn(String profileName) {
-    Selenide.$("#coding-rules-detail-quality-profiles").shouldHave(text(profileName));
+    $("#coding-rules-detail-quality-profiles").shouldHave(text(profileName));
     return this;
   }
 
+  public RuleDetails shouldHaveTotalIssues(int issues) {
+    $(".js-rule-issues h3").shouldHave(text(String.valueOf(issues)));
+    return this;
+  }
+
+  public RuleDetails shouldHaveIssuesOnProject(String projectName, int issues) {
+    $(".coding-rules-most-violated-projects").shouldHave(
+      Condition.and("", text(projectName), text(String.valueOf(issues))));
+    return this;
+  }
+
+  public ExtendedDescription extendDescription() {
+    return new ExtendedDescription().start();
+  }
+
+  public class ExtendedDescription {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make this a "static" inner class. ([squid:S2694](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fit-rules&issues=AWEuYoNKzwlWAzALw-9m&open=AWEuYoNKzwlWAzALw-9m)

[ISSUE_KEY]: AWEuYoNKzwlWAzALw-9m